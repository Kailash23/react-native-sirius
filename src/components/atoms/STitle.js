import React from 'react';
import { StyleSheet } from 'react-native';
import { Title, withTheme } from 'react-native-paper';
import { customFonts } from '@styles/customFonts';
import {
  SIZES,
  HalfCardImageWidth,
  getFontSize,
  HalfCardImageHeight
} from '@styles/sizes';

function STitle(props) {
  const {
    size,
    title,
    paddingTop,
    paddingBottom,
    style,
    half,
    semiHalf,
    paddingHorizontal,
    lineHeight,
    fontFamily,
    titleColor,
    shadows,
    uppercase,
    avoidTextScaling,
    numberOfLines,
    fontWeight,
    opacity
  } = props;

  const { colors } = props.theme;

  function scaledText(size) {
    if (avoidTextScaling) {
      return size;
    } else {
      return getFontSize(size);
    }
  }

  return (
    <Title
      numberOfLines={numberOfLines && numberOfLines}
      selectable={true}
      style={[
        shadows ? styles.shadows : null,
        {
          fontFamily: fontFamily ? fontFamily : customFonts.torstarDecSemiBold,
          fontSize: size ? scaledText(size) : scaledText(SIZES.FONT_LARGE_1),
          paddingTop: paddingTop ? paddingTop : 0,
          paddingBottom: paddingBottom ? paddingBottom : 0,
          paddingHorizontal: paddingHorizontal ? paddingHorizontal : 0,
          width: half
            ? HalfCardImageWidth * 0.9
            : semiHalf
            ? HalfCardImageWidth * 0.65
            : '99%',
          color: titleColor ? titleColor : colors.text,
          textTransform: uppercase ? 'uppercase' : null,
          margin: 0,
          lineHeight: lineHeight
            ? scaledText(lineHeight)
            : scaledText(SIZES.LINE_HEIGHT),
          fontWeight: fontWeight ? fontWeight : '400',
          opacity: opacity ? opacity : 1
        },
        { ...style }
      ]}>
      {title}
    </Title>
  );
}

const styles = StyleSheet.create({
  shadows: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1
    },
    shadowOpacity: 0.9,
    shadowRadius: 1.41
  }
});

export default withTheme(STitle);
