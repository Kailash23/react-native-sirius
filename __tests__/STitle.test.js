/**
 * @jest-environment jsdom
 */
import STitle from '@atoms/STitle';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Text } from 'react-native';

describe('Testing <STitle/>', () => {
  test('STitle should render properly with given title', () => {
    const wrapper = shallow(
      <STitle
        title={'Test Title'}
        size={20}
        lineHeight={29}
        style={{
          color: 'black',
          textAlign: 'center'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('STitle should render half width', () => {
    const wrapper = shallow(
      <STitle
        half
        title={'Test Title'}
        size={20}
        lineHeight={29}
        style={{
          color: 'black',
          textAlign: 'center'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('STitle should render semi-half width', () => {
    const wrapper = shallow(
      <STitle
        semiHalf
        title={'Test Title'}
        size={20}
        lineHeight={29}
        style={{
          color: 'black',
          textAlign: 'center'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
