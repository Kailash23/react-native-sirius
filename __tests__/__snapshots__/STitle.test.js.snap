// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <STitle/> STitle should render half width 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  half={true}
  lineHeight={29}
  size={20}
  style={
    Object {
      "color": "black",
      "textAlign": "center",
    }
  }
  title="Test Title"
/>
`;

exports[`Testing <STitle/> STitle should render properly with given title 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  lineHeight={29}
  size={20}
  style={
    Object {
      "color": "black",
      "textAlign": "center",
    }
  }
  title="Test Title"
/>
`;

exports[`Testing <STitle/> STitle should render semi-half width 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  lineHeight={29}
  semiHalf={true}
  size={20}
  style={
    Object {
      "color": "black",
      "textAlign": "center",
    }
  }
  title="Test Title"
/>
`;
